### PR TITLE
Setting minimum clang version

### PR DESCRIPTION
**Pull request by roigcarlo**
**📝 Description**
Sets minimum clang version to 12.

For context read [#10911](https://github.com/KratosMultiphysics/Kratos/pull/10911#issuecomment-1525451946)

**🆕 Changelog**
- Setting minimum clang version to 12.0

### CODE REVIEW

**Comment by loumalouomega** [comment]
It is quite recent version, but well...

**Review by philbucher — DISMISSED** [review]
nice!

File: CMakeLists.txt [hunk 1]
```diff
@@ -191,13 +191,19 @@ message(STATUS "CMAKE_CXX_COMPILER_ID = ${CMAKE_CXX_COMPILER_ID}")
 if(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
       if(${CMAKE_CXX_COMPILER_ID} MATCHES Clang)
+        if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 12.0)
+          message( FATAL_ERROR "Please use Version 12.0 or greater")
+        endif(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 12.0)
```
Comment (philbucher) [inline CMakeLists.txt]: how about moving this check outside this anyway super nasty section?

Then you also dont need to duplicate it

**Comment by roigcarlo** [comment]
> It is quite recent version, but well...

its from 2 years ago...

File: CMakeLists.txt (hunk 1 above)
Comment (roigcarlo) [reply CMakeLists.txt]: yep you are right....

**Comment by philbucher** [comment]
> > It is quite recent version, but well...
> 
> its from 2 years ago...

lets upgrade and potentially do sth if someone complains

File: CMakeLists.txt [hunk 2]
```diff
@@ -186,24 +186,25 @@ if(${CMAKE_COMPILER_IS_GNUCXX})
   message(STATUS "Additional default options were set for GCC")
 endif(${CMAKE_COMPILER_IS_GNUCXX})
 
+# Check for correct clang version
+if(${CMAKE_CXX_COMPILER_ID} MATCHES Clang)
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 12.0)
+    message( FATAL_ERROR "Please use Version 12.0 or greater")
+  endif(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 12.0)
+endif(${CMAKE_CXX_COMPILER_ID} MATCHES Clang)
```
Comment (philbucher) [inline CMakeLists.txt]:
```suggestion
if(${CMAKE_CXX_COMPILER_ID} MATCHES Clang AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 12.0)
    message( FATAL_ERROR "Please use Version 12.0 or greater")
endif()
```

**Review by philbucher — DISMISSED** [review]

**Review by philbucher — APPROVED** [review]

**Pull request merged**